src/Symfony/Component/Console/SignalRegistry/SignalMap.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\SignalRegistry;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class SignalMap
+{
+    private static array $map;
+
+    public static function getSignalName(int $signal): ?string
+    {
+        if (!isset(self::$map)) {
+            $r = new \ReflectionExtension('pcntl');
```

Comment:
what if `pcntl` is not loaded (for instance when running on windows) ? Should we initialize the map to an empty array in that case ?

Reply:
I didn't know if I should throw an exception or return null.
WDYT?

Reply:
given that it already returns null for unknown signals, I would return null.

Reply:
Btw, the test about the unknown signal could be extracted into a separate test (instead of being in the same dataprovider), and the `@requires extension pcntl` moved to the test covering known signals only.

Reply:
Good idea 👍🏼 
I updated the PR